### ISSUE TITLE
Handle login requests for multiple tenants in the same browser

### ISSUE THREAD

**Issue by ayshsandu** [wso2/product-is#5365]
Consider the following scenario.

In the IS there are two tenants, carbon.super and abc.com.
Each tenant has Service Providers SP1 and SP2 (non-SaaS apps) corresponding order. 

Now a user logs in to SP1 app in super.tenant. 
In the same browser, a user tries to access SP2 in the abc.com tenant. What is the expected behaviour? 

As of now, we get the following error page.
![image](https://user-images.githubusercontent.com/5061791/58863978-876f0e80-86d1-11e9-91eb-1765195c1aed.png)

And below error can be observed in the console of latests pack 

```
[2019-04-24 10:51:11,749] ERROR {org.wso2.carbon.identity.application.authentication.framework.handler.request.impl.DefaultRequestCoordinator} -  Exception in Authentication Framework

org.wso2.carbon.identity.application.authentication.framework.exception.FrameworkException: Service Provider tenant domain must be equal to user tenant domain for non-SaaS applications

	at org.wso2.carbon.identity.application.authentication.framework.handler.request.impl.DefaultAuthenticationRequestHandler.concludeFlow(DefaultAuthenticationRequestHandler.java:289)

	at org.wso2.carbon.identity.application.authentication.framework.handler.request.impl.DefaultAuthenticationRequestHandler.handle(DefaultAuthenticationRequestHandler.java:145)
```

I think this is how we have implemented so far to, invalidate login requests from secondary tenant's SPs when there is already a logged in session for a particular tenant?

However, once an application reaches the above state, the user is unable to recover from that state for the SPs in secondary tenants until log-out from the first tenant or the current session time outs. There can be situations where end users are not at all aware of tenancy behind the server and just trying to consume the applications.

**Comment by abhi4u1947** [comment]
This becomes a very important use case for some organizations offering multi-tenant systems and persons working for multiple tenants apps... 

Like someone working as client of an organization-x and organization-y as an auditor and there is an audit application offered by the main organization offering audit application as a multi-tenant system.

**Comment by FinalFBoss** [comment]
Hi,
I've done some little changes on 'org.wso2.carbon.identity.application.authentication.framework' (5.17.5) for customize the auth cookie 'commonAuthId' respect of the 'tenantDomain'. 
Example: 
- if the tenantDomain is 'test', the new name cookie is 'commonAuthId-test'.
- if the tenantDomain is 'carbon.super', the new cookie name is 'commong
With this changes the user can login into different Identity Provider of different tenantDomain at the same time.

If it's correct, how can I send this contribution ?

**Comment by IsuraD** [comment]
related to https://github.com/wso2/product-is/issues/9478

**Comment by pulasthi7** [comment]
Addressed along with wso2/product-is#8975
